src/main/java/io/cdap/plugin/http/source/common/BaseHttpSourceConfig.java

```
@@ -0,0 +1,867 @@
+  }
+
+  @Nullable
+  public String getKeystoreFile() {
+    return keystoreFile;
+  }
+
+  @Nullable
+  public KeyStoreType getKeystoreType() {
+    return ConfigUtil.getEnumValueByString(KeyStoreType.class, keystoreType, PROPERTY_KEYSTORE_TYPE);
+  }
+
+  @Nullable
+  public String getKeystorePassword() {
+    return keystorePassword;
+  }
+
+  @Nullable
+  public String getKeystoreKeyAlgorithm() {
+    return keystoreKeyAlgorithm;
+  }
+
+  @Nullable
+  public String getTrustStoreFile() {
+    return trustStoreFile;
+  }
+
+  @Nullable
+  public KeyStoreType getTrustStoreType() {
+    return ConfigUtil.getEnumValueByString(KeyStoreType.class, trustStoreType, PROPERTY_TRUSTSTORE_TYPE);
+  }
+
+  @Nullable
+  public String getTrustStorePassword() {
+    return trustStorePassword;
+  }
+
+  @Nullable
+  public String getTrustStoreKeyAlgorithm() {
+    return trustStoreKeyAlgorithm;
+  }
+
+  @Nullable
+  public String getTransportProtocols() {
+    return transportProtocols;
+  }
+
+  @Nullable
+  public String getCipherSuites() {
+    return cipherSuites;
+  }
+
+  @Nullable
+  public Schema getSchema() {
+    try {
+      return Strings.isNullOrEmpty(schema) ? null : Schema.parseJson(schema);
+    } catch (IOException e) {
+      throw new InvalidConfigPropertyException("Unable to parse output schema: " +
+                                                 schema, e, PROPERTY_SCHEMA);
+    }
+  }
+
+  @Nullable
+  public Map<String, String> getHeadersMap() {
+    return ConfigUtil.getMapFromKeyValueString(headers);
+  }
+
+  @Nonnull
+  public Map<Pattern, HttpErrorHandlingStrategy> getHttpErrorsHandlingMap() {
+    Map<String, String> httpErrorsHandlingMap = ConfigUtil.getMapFromKeyValueString(httpErrorsHandling);
+    Map<Pattern, HttpErrorHandlingStrategy> result = new LinkedHashMap<>(); // order of pattern checks is important
+
+    for (Map.Entry<String, String> entry : httpErrorsHandlingMap.entrySet()) {
+      result.put(Pattern.compile(entry.getKey()),
```

if the pattern is invalid, we should throw an exception with a nice message.

---

done